android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/SwipeDismissListViewTouchListener.java

```diff
@@ -0,0 +1,356 @@
+     */
+    public interface DismissCallbacks {
+        /**
+         * Called to determine whether the given position can be dismissed.
+         */
+        boolean canDismiss(int position);
+
+        /**
+         * Called when the user has indicated they she would like to dismiss one or more list item
+         * positions.
+         *
+         * @param listView               The originating {@link ListView}.
+         * @param position               Position of listview dismiss
+         */
+        void onDismiss(ListView listView, int position);
+    }
+
+    /**
+     * Constructs a new swipe-to-dismiss touch listener for the given list view.
+     *
+     * @param listView  The list view whose items should be dismissable.
+     * @param callbacks The callback to trigger when the user has indicated that she would like to
+     *                  dismiss one or more list items.
+     */
+    public SwipeDismissListViewTouchListener(ListView listView, DismissCallbacks callbacks) {
+        ViewConfiguration vc = ViewConfiguration.get(listView.getContext());
+        mSlop = vc.getScaledTouchSlop();
+        mMinFlingVelocity = vc.getScaledMinimumFlingVelocity() * 16;
+        mMaxFlingVelocity = vc.getScaledMaximumFlingVelocity();
+        mAnimationTime = listView.getContext().getResources().getInteger(
+                android.R.integer.config_shortAnimTime);
+        mListView = listView;
+        mCallbacks = callbacks;
+    }
+
+    /**
+     * Enables or disables (pauses or resumes) watching for swipe-to-dismiss gestures.
+     *
+     * @param enabled Whether or not to watch for gestures.
+     */
+    public void setEnabled(boolean enabled) {
+        mPaused = !enabled;
+    }
+
+    public static boolean isLocked() {
+        return mSwiping || mAnimating;
+    }
+
+    /**
+     * Returns an {@link AbsListView.OnScrollListener} to be added to the {@link
+     * ListView} using {@link ListView#setOnScrollListener(AbsListView.OnScrollListener)}.
+     * If a scroll listener is already assigned, the caller should still pass scroll changes through
+     * to this listener. This will ensure that this {@link SwipeDismissListViewTouchListener} is
+     * paused during list view scrolling.
+     *
+     * @see SwipeDismissListViewTouchListener
+     */
+    public AbsListView.OnScrollListener makeScrollListener() {
+        return new AbsListView.OnScrollListener() {
+            @Override
+            public void onScrollStateChanged(AbsListView absListView, int scrollState) {
+                setEnabled(scrollState != AbsListView.OnScrollListener.SCROLL_STATE_TOUCH_SCROLL);
+            }
+
+            @Override
+            public void onScroll(AbsListView absListView, int i, int i1, int i2) {
+            }
+        };
+    }
+
+    @Override
+    public boolean onTouch(View view, MotionEvent motionEvent) {
+        if (mViewWidth < 2) {
+            mViewWidth = mListView.getWidth();
```

Comment:
I don't think we need to cache this value, it's fine to get it each time you enter onTouch.  Caching the width could lead to weird behavior on screen rotation.